src/FormCheck.tsx

```diff
@@ -81,9 +83,26 @@ const propTypes = {
    */
   children: PropTypes.node,
 
+  /**
+   * Groups controls horizontally with other `FormCheck`s.
+   */
   inline: PropTypes.bool,
+
+  /**
+   * Disables the control.
+   */
   disabled: PropTypes.bool,
+
+  /**
+   * `title` attribute for the underlying `FormCheckLabel`.
+   */
   title: PropTypes.string,
+
+  /**
+   * Label for the control.
+   *
+   * This is **required** when `type="switch"`.
+   */
   label: PropTypes.node,
```

Comment:
A label shouldn't actually be required to use the switch. The issue is we don't render a `label` element at all when it's not provided. More correctly we should render an empty label for the switch case I think.